dockers/texcmp/texcmp.js

```diff
@@ -0,0 +1,244 @@
+                var bi = browser.imag.get(y, x);
+                var lr = latex.real.get(y, x);
+                var li = latex.imag.get(y, x);
+                real.set(y, x, br * lr + bi * li);
+                imag.set(y, x, br * li - bi * lr);
+            }
+        }
+
+        // Step 6b: (real + i*imag) = inverseFFT(real + i*imag)
+        fft(-1, real, imag);
+
+        // Step 6c: find position where the (squared) absolute value is maximal
+        var ox = 0;
+        var oy = 0;
+        var ov = -1;
+        for (y = 0; y < alignHeight; ++y) {
+            for (x = 0; x < alignWidth; ++x) {
+                var or = real.get(y, x);
+                var oi = imag.get(y, x);
+                var sq = or * or + oi * oi;
+                if (ov < sq) {
+                    ov = sq;
+                    ox = x;
+                    oy = y;
+                }
+            }
+        }
+
+        // Step 6d: Treat negative offsets in a non-cyclic way
+        if (oy > (alignHeight >>> 1)) {
+            oy -= alignHeight;
+        }
+        if (ox > (alignWidth >>> 1)) {
+            ox -= alignWidth;
+        }
+        console.log("Positioned " + key + ": " + ox + ", " + oy);
+
+        // Step 7: use these offsets to compute difference illustration
+        var uw = Math.max(browser.width, latex.width) + Math.abs(ox);
+        var uh = Math.max(browser.height, latex.height) + Math.abs(oy);
+        var bx = Math.max(ox, 0);
+        var by = Math.max(oy, 0);
+        var lx = Math.max(-ox, 0);
+        var ly = Math.max(-oy, 0);
+        return execFile("convert", [
+            "(", pngFile, "-grayscale", "Rec709Luminance",
+            "-extent", uw + "x" + uh + "-" + lx + "-" + ly,
+            ")",
+            "(", browserFile, "-grayscale", "Rec709Luminance",
+            "-extent", uw + "x" + uh + "-" + bx + "-" + by,
+            ")",
+            "(", "-clone", "0-1", "-compose", "darken", "-composite", ")",
+            "-channel", "RGB", "-combine", "-trim", diffFile
+        ]);
+    }).then(function() {
+        console.log("Compared " + key);
+    });
+};
+
+// Create a directory, but ignore error if the directory already exists.
+function mkdir(dir) {
+    return Q.nfcall(fs.mkdir, dir).catch(function(err) {
+        if (err.code !== "EEXIST")
+            throw err;
+    });
+}
+
+// Execute a given command, and return a promise to its output.
+function execFile(cmd, args, opts) {
+    var deferred = Q.defer();
+    child_process.execFile(cmd, args, opts, function(err, stdout, stderr) {
+        if (err) {
+            console.error("Error executing " + cmd + " " + args.join(" "));
+            console.error(stdout + stderr);
+            err.stdout = stdout;
+            err.stderr = stderr;
+            deferred.reject(err);
+        } else {
+            deferred.resolve(stdout);
+        }
+    });
+    return deferred.promise;
+}
+
+// Read given file, and apply FFT transformation to it.
+function fftPNG(file) {
```

Comment:
That's okay.  Could you instead inline ```parsed``` so that we're not relying on hoisting for this work?  It's weird to have code appear after a ```return```.